tests/integration/cli/commands/test_issue_2268.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+import os.path
+import sys
+
+import pytest
+
+from pex.compatibility import commonpath
+from pex.resolve.resolver_configuration import ResolverVersion
+from pex.typing import TYPE_CHECKING
+from testing import run_pex_command
+from testing.cli import run_pex3
+
+if TYPE_CHECKING:
+    from typing import Any
+
+
+@pytest.mark.skipif(
+    sys.version_info[:2] < (3, 5),
+    reason=(
+        "This test relies on the python-forge 18.6.0 offering only the single "
+        "python_forge-18.6.0-py35-none-any.whl distribution on PyPI and this wheel only works for "
+        "Python 3.5 and greater"
+    ),
+)
+@pytest.mark.parametrize(
+    "resolver_version",
+    [
+        pytest.param(resolver_version, id=str(resolver_version))
+        for resolver_version in ResolverVersion.values()
+        if ResolverVersion.applies(resolver_version)
+    ],
+)
+def test_abi_none_locking(
+    tmpdir,  # type: Any
+    resolver_version,  # type: ResolverVersion.Value
+):
+    # type: (...) -> None
+
+    pex_root = os.path.join(str(tmpdir), "pex_root")
+    lock = os.path.join(str(tmpdir), "lock.json")
+    run_pex3(
+        "lock",
+        "create",
+        "--style",
+        "universal",
+        "--resolver-version",
+        str(resolver_version),
+        "--interpreter-constraint",
+        "==3.11.*",
+        "python-forge==18.6.0",
+        "-o",
+        lock,
+        "--indent",
+        "2",
+        "--pex-root",
+        pex_root,
+    ).assert_success()
```

Comment:
Prior to the fix above, this test fails as expected (in the same way the real case failed for the issue reporter:
```
$ ./dtox.sh -epy311-integration -- --devpi -vvsk test_issue_2268 -n2
...
=================================== FAILURES ===================================
__________________ test_abi_none_locking[pip-legacy-resolver] __________________
[gw0] linux -- Python 3.11.6 /development/pex/.tox/py311-integration/bin/python
...
E       AssertionError: integration test failed: return_code=1, output=, error=pid 877 -> /tmp/pytest-of-jsirois/pytest-13/popen-gw0/test_abi_none_locking_pip_lega0/pex_root/venvs/f646e8ec93937037a1a0dd448a5db57db632bd73/a21ef60f2b1adcb6fb7c6a50aa0e7e07a678e7bc/bin/python -sE /tmp/pytest-of-jsirois/pytest-13/popen-gw0/test_abi_none_locking_pip_lega0/pex_root/venvs/f646e8ec93937037a1a0dd448a5db57db632bd73/a21ef60f2b1adcb6fb7c6a50aa0e7e07a678e7bc/pex --disable-pip-version-check --no-python-version-warning --exists-action a --no-input --use-deprecated legacy-resolver -q --cache-dir /tmp/pytest-of-jsirois/pytest-13/popen-gw0/test_abi_none_locking_pip_lega0/pex_root/pip/20.3.4-patched/pip_cache --log /tmp/pex-pip-log.cc9l_rd5/pip.log download --dest /tmp/tmpn098tmyx/development.pex..tox.py311-integration.bin.python python-forge==18.6.0 --retries 5 --timeout 15 exited with 1 and STDERR:
E       ERROR: Could not find a version that satisfies the requirement python-forge==18.6.0 (from versions: none)
E       ERROR: No matching distribution found for python-forge==18.6.0

testing/__init__.py:374: AssertionError
___________________ test_abi_none_locking[pip-2020-resolver] ___________________
[gw1] linux -- Python 3.11.6 /development/pex/.tox/py311-integration/bin/python
...
E       AssertionError: integration test failed: return_code=1, output=, error=pid 878 -> /tmp/pytest-of-jsirois/pytest-13/popen-gw1/test_abi_none_locking_pip_20200/pex_root/venvs/f646e8ec93937037a1a0dd448a5db57db632bd73/a21ef60f2b1adcb6fb7c6a50aa0e7e07a678e7bc/bin/python -sE /tmp/pytest-of-jsirois/pytest-13/popen-gw1/test_abi_none_locking_pip_20200/pex_root/venvs/f646e8ec93937037a1a0dd448a5db57db632bd73/a21ef60f2b1adcb6fb7c6a50aa0e7e07a678e7bc/pex --disable-pip-version-check --no-python-version-warning --exists-action a --no-input -q --cache-dir /tmp/pytest-of-jsirois/pytest-13/popen-gw1/test_abi_none_locking_pip_20200/pex_root/pip/20.3.4-patched/pip_cache --log /tmp/pex-pip-log.xdty2u9h/pip.log download --dest /tmp/tmpaoej4mji/development.pex..tox.py311-integration.bin.python python-forge==18.6.0 --retries 5 --timeout 15 exited with 1 and STDERR:
E       ERROR: Could not find a version that satisfies the requirement python-forge==18.6.0
E       ERROR: No matching distribution found for python-forge==18.6.0

testing/__init__.py:374: AssertionError
=========================== short test summary info ============================
FAILED tests/integration/cli/commands/test_issue_2268.py::test_abi_none_locking[pip-legacy-resolver] - AssertionError: integration test failed: return_code=1, output=, error=pid ...
FAILED tests/integration/cli/commands/test_issue_2268.py::test_abi_none_locking[pip-2020-resolver] - AssertionError: integration test failed: return_code=1, output=, error=pid ...
============================== 2 failed in 4.64s ===============================
```